crates/libcontainer/src/process/container_init_process.rs

```diff
@@ -741,6 +750,59 @@ fn set_io_priority(syscall: &dyn Syscall, io_priority_op: &Option<LinuxIOPriorit
     Ok(())
 }
 
+/// Set the RT priority of a thread
+fn setup_scheduler(sc_op: &Option<Scheduler>) -> Result<()> {
+    if let Some(sc) = sc_op {
+        let policy: u32 = match *sc.policy() {
+            LinuxSchedulerPolicy::SchedOther => 0,
+            LinuxSchedulerPolicy::SchedFifo => 1,
+            LinuxSchedulerPolicy::SchedRr => 2,
+            LinuxSchedulerPolicy::SchedBatch => 3,
+            LinuxSchedulerPolicy::SchedIso => 4,
+            LinuxSchedulerPolicy::SchedIdle => 5,
+            LinuxSchedulerPolicy::SchedDeadline => 6,
+        };
+        let mut flags_value: u64 = 0;
+        if let Some(flags) = sc.flags() {
+            for flag in flags {
+                match *flag {
+                    LinuxSchedulerFlag::SchedResetOnFork => flags_value |= 0x01,
+                    LinuxSchedulerFlag::SchedFlagReclaim => flags_value |= 0x02,
+                    LinuxSchedulerFlag::SchedFlagDLOverrun => flags_value |= 0x04,
+                    LinuxSchedulerFlag::SchedFlagKeepPolicy => flags_value |= 0x08,
+                    LinuxSchedulerFlag::SchedFlagKeepParams => flags_value |= 0x10,
+                    LinuxSchedulerFlag::SchedFlagUtilClampMin => flags_value |= 0x20,
+                    LinuxSchedulerFlag::SchedFlagUtilClampMax => flags_value |= 0x40,
+                }
+            }
+        }
+        let mut a = nc::sched_attr_t {
+            size: mem::size_of::<nc::sched_attr_t>().try_into().unwrap(),
+            sched_policy: policy,
+            sched_flags: flags_value,
+            sched_nice: sc.nice().unwrap_or(0),
+            sched_priority: sc.priority().unwrap_or(0) as u32,
+            sched_runtime: sc.runtime().unwrap_or(0),
+            sched_deadline: sc.deadline().unwrap_or(0),
+            sched_period: sc.period().unwrap_or(0),
+            sched_util_min: 0,
+            sched_util_max: 0,
+        };
+        // TODO when nix or libc support this function, replace nx crates.
+        unsafe {
+            let result = nc::sched_setattr(1, &mut a, 0);
```

Comment:
Hey, I was looking through the runc implementation for this, [here](https://github.com/opencontainers/runc/pull/4025/files#diff-84fffb149f8b86c903e09613cbebad12de596f4139d54e288388b1458e08b8ec) , and one difference I see is that in their setattr call, the first param, i.e. pid is set to 0, whereas here we are setting it to 1. Maybe that is causing some differences, resulting in failure. Any idea why the 0 v/s 1 difference is there?

Reply:
Here it is 0, it was 0 at first, but the runc test failed, so I tried to change this parameter to 1, but forgot to change it back.